packages/ckeditor5-table/src/tablecaption/toggletablecaptioncommand.js

```diff
@@ -114,7 +114,8 @@ export default class ToggleTableCaptionCommand extends Command {
 		// Store the caption content so it can be restored quickly if the user changes their mind.
 		tableCaptionEditing._saveCaption( tableElement, captionElement );
 
+		writer.setSelection( captionElement, 'on' );
+		model.deleteContent( model.document.selection );
```

Comment:
`model.deleteContent()` should be able to take `Range` instance as the parameter. Did you try `model.deleteContent( model.createRangeOn( captionElement ) )`? Was it failing for some reason?

Reply:
In this case, have you tried `model.deleteContent( model.createSelection( captionElement, 'on' ) )`?